src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2050,12 +2054,12 @@ private BoundNode VisitLambdaOrLocalFunction(IBoundLambdaOrFunction node)
             ImmutableArray<PendingBranch> pendingReturns = RemoveReturns();
             RestorePending(oldPending);
             IntersectWith(ref finalState, ref this.State); // a no-op except in region analysis
-            this.State.ResultType = _invalidType;
+            _result = _invalidType;
```

Comment:
Should `_result` be restored before exiting this method?
More generally, is it correct to say that a `Visit` method should never return with `_result` let in unset state? If that's correct, is there a common place we could assert/enforce that?

Reply:
I'll address this specific case in a separate PR for lambda conversions. For other expressions, `VisitExpressionWithoutStackGuard` asserts that `_result` is set to a type that matches `node.Type`.